lasagne/tests/layers/test_helper.py

```diff
@@ -718,6 +718,29 @@ def test_get_all_params(self):
                 (l2.get_params(regularizable=True) +
                  l3.get_params(regularizable=True)))
 
+    def test_get_all_params_with_unwrap_shared(self):
+        from lasagne.layers import (InputLayer, DenseLayer, get_all_params)
+        from theano.tensor.var import TensorVariable
+        l1 = InputLayer((10, 20))
+        l2 = DenseLayer(l1, 30)
+        l3 = DenseLayer(l2, 40)
+
+        assert get_all_params(l3) == get_all_params(l3, unwrap_shared=False)
+
+        l2.add_param((l2.W).dot(l2.W.T), (20, 20), name="W.WT")
```

Comment:
I'd prefer to create an expression for `W` of `l3` before creating `l3`. Maybe:
```python
W1 = theano.shared(floatX(np.zeros((30,2))))
W2 = theano.shared(floatX(np.zeros((2,40))))
W_expr = T.dot(W1, W2)
l3 = DenseLayer(l2, W=W_expr)
```
Then directly check that the correct instances are returned (in correct order!) with different settings for `unwrap_shared`, just using two assertions.